sz/CMakeLists.txt

```diff
@@ -92,11 +92,18 @@ set_target_properties (SZ PROPERTIES
     OUTPUT_NAME_RELWITHDEBINFO ${LIB_RELEASE_NAME}
   )
 
-target_link_libraries (SZ PUBLIC ${ZLIB_dep} ${ZSTD_dep} m)
+if (WIN32 AND NOT MINGW)
+    target_link_libraries (SZ PUBLIC ${ZLIB_dep} ${ZSTD_dep})
+else ()
+    target_link_libraries (SZ PUBLIC ${ZLIB_dep} ${ZSTD_dep} m)
+endif ()
 
-target_compile_options(SZ
-	PRIVATE $<$<CONFIG:Debug>:-Wall -Wextra -Wpedantic -Wno-unused-parameter>
-	)
+if(CMAKE_COMPILER_IS_GNUCXX)
```

Comment:
This is probably overly restrictive.  We want to add these flags for Clang too.  We can either check for the flags with CheckCXXCompilerFlag https://cmake.org/cmake/help/latest/module/CheckCXXCompilerFlag.html, or we can change this to use CMAKE_<LANG>_COMPILER_ID to check if the compiler is not MSVC.  I would prefer the former.

Reply:
Since at other places there were checks for WIN32 without MINGW, I have used the same check.

It should not harm any other platform.